packages/core/src/lib/sign-in-experience.test.ts

```diff
@@ -59,6 +71,58 @@ describe('validate branding', () => {
   });
 });
 
+describe('validate language info', () => {
+  it('should call findAllCustomLanguageKeys', async () => {
+    await validateLanguageInfo({
+      autoDetect: true,
+      fallbackLanguage: 'zh-CN',
+      fixedLanguage: 'en',
+    });
+    expect(findAllCustomLanguageKeys).toBeCalledTimes(1);
+  });
+
+  it('should pass when the language is built-in supported', async () => {
+    const builtInSupportedLanguage = 'tr-TR';
+    await expect(
+      validateLanguageInfo({
+        autoDetect: true,
+        fallbackLanguage: builtInSupportedLanguage,
+        fixedLanguage: 'en',
+      })
+    ).resolves.not.toThrow();
+    expect(findAllCustomLanguageKeys).toBeCalledTimes(1);
+  });
+
+  it('should pass when the language is custom supported', async () => {
+    const customSupportedLanguage = 'zh-HK';
```

Comment:
Hard to track this code if the `zh-HK` becomes our built-in language.
Maybe add an assertion to check 'unsupported' languages.

Reply:
Add the statement assert that `zh-HK` is custom only language (not in built-in languages)
```ts
    const customOnlySupportedLanguage = 'zh-HK';
    expect(customOnlySupportedLanguage in builtInLanguages).toBeFalsy();
    findAllCustomLanguageKeys.mockResolvedValueOnce([customOnlySupportedLanguage]);
```